core/src/impl/Kokkos_ViewMapping.hpp

```diff
@@ -2892,25 +2892,29 @@ struct ViewValueFunctor;
 
 template <class DeviceType, class ValueType>
 struct ViewValueFunctor<DeviceType, ValueType, false /* is_scalar */> {
-  using ExecSpace  = typename DeviceType::execution_space;
-  using PolicyType = Kokkos::RangePolicy<ExecSpace, Kokkos::IndexType<int64_t>>;
+  using ExecSpace = typename DeviceType::execution_space;
+
+  struct DestroyTag {};
+  struct ConstructTag {};
 
   ExecSpace space;
   ValueType* ptr;
   size_t n;
-  bool destroy;
   std::string name;
   bool default_exec_space;
 
+  template <class _ValueType = ValueType>
   KOKKOS_INLINE_FUNCTION
-  void operator()(const size_t i) const {
-    if (destroy) {
-      (ptr + i)->~ValueType();
-    }  // KOKKOS_IMPL_CUDA_CLANG_WORKAROUND this line causes ptax error
-       // __cxa_begin_catch in nested_view unit-test
-    else {
-      new (ptr + i) ValueType();
-    }
+      std::enable_if_t<std::is_default_constructible<_ValueType>::value>
+      operator()(ConstructTag const&, const size_t i) const {
+    new (ptr + i) ValueType();
+  }
+
+  KOKKOS_INLINE_FUNCTION void operator()(DestroyTag const&,
+                                         const size_t i) const {
+    // KOKKOS_IMPL_CUDA_CLANG_WORKAROUND this line causes ptax error
```

Comment:
does this comment still has any meaning? Not holding this up because of it but would be good to figure out. I also see that one of hte tests is commented out for HIP too ?

Reply:
As far as I can see, the test is also enabled for Cuda+Clang so I removed the comment.